lib/jit_preloader/active_record/associations/preloader/ar7_branch.rb

```diff
@@ -0,0 +1,24 @@
+module JitPreloader
+  module PreloaderBranch
+    """
+    From ActiveRecord version >= 7.x.x, it introduces a improvement for preloading associations in batch
+    https://github.com/rails/rails/blob/main/activerecord/lib/active_record/associations/preloader.rb#L121
+
+    Previously we make sure to check and direct return if an association klass is not a valid ActiveRecord::Base
+    class through prepended methods within lib/jit_preloader/active_record/associations/preloader/arx_association.rb
+    The Batch approach move the check of association klass forward before calling individual association.
+    https://github.com/rails/rails/blob/main/activerecord/lib/active_record/associations/preloader/batch.rb#L17
+
+    This module prepend on Branch class which is the object we grab the list of loaders aka associations
+    and remove the one that does not have a valid ActiveRecord::Base klass
+    """
+
+    def loaders
```

Comment:
So we're changing rails behaviour here, right?  Like, there's a super method, that rails might expect to behave a certain way - what breaks?

Reply:
The behavior we still want to enforce on version > 7. xx is this [spec scenario](https://github.com/clio/jit_preloader/blob/master/spec/lib/jit_preloader/preloader_spec.rb#L158-L168), which still load the rest of records and ignore the one that contains invalid association klass.

I also enforce that on versions 5.xx and 6.xx with https://github.com/clio/jit_preloader/blob/master/lib/jit_preloader/active_record/associations/preloader/ar5_association.rb#L21 and https://github.com/clio/jit_preloader/blob/master/lib/jit_preloader/active_record/associations/preloader/ar6_association.rb by checking the klass is a child of `ActiveRecord::Base` before passing to the its parent.

Unfortunately, https://github.com/clio/jit_preloader/blob/master/lib/jit_preloader/active_record/associations/preloader/ar7_association.rb does not work due to the change of using Batch association runner in version 7.xx

Reply:
Is there a way we can do this without monkey-patching rails, and instead keeping the behaviour for our gem inside our gem?

Reply:
I don't see a good way to avoid a monkey-patch here. We have an existing monkey-patch, and the new code organization in AR 7 means that monkey-patch isn't going to get hit at the right time. We can't implement this behaviour without monkey-patches. This three-line method seems like the least bad solution.